pkg/services/object/get/prm.go

```diff
@@ -21,6 +22,30 @@ type RangePrm struct {
 	rng *object.Range
 }
 
+var (
+	errRangeZeroLength = errors.New("zero range length")
+	errRangeOverflow   = errors.New("range overflow")
+)
+
+// Validate pre-validates `OBJECTRANGE` request's parameters content
+// without access to the requested object's payload.
+func (p RangePrm) Validate() error {
+	if p.rng != nil {
+		off := p.rng.GetOffset()
+		l := p.rng.GetLength()
+
+		if l == 0 {
+			return errRangeZeroLength
+		}
+
+		if off <= off+l {
+			return errRangeOverflow
+		}
```

Comment:
Isn't it always true for positive lengths?

Reply:
a mistake, fixed